backend/src/services/members/db.ts

```diff
@@ -0,0 +1,23 @@
+import knex from '../../database';
+
+type stil_id = string;
+
+interface IDbMember {
+  stil_id: stil_id,
+  name: string,
+  programme: string,
+  first_year: number,
+}
+
+const getMember = (stil_id: stil_id): Promise<IDbMember | undefined> => {
+  return knex<IDbMember>('members')
+    .select('*')
+    .where({stil_id: stil_id})
+    .first()
+    .catch((reason: any) => undefined)
```

Comment:
Är undefined rätt värde här? Varför inte null?

Reply:
Jag är lite osäker på hur undefined och null ska användas ideomatiskt i typescript/javascript. De får identisk effekt i koden är jag rätt säker på.

Reply:
Undefined är värdet på en variabel eller ett attribut som inte tilldelats något värde överhuvudtaget. Null är värdet som berättar att det explicit saknas ett värde för saken som variabeln representerar. Tänk lite pekarvariabel som är oinitialiserad vs pekarvariabel satt till null, i lågnivåtermer. Så du vill i princip aldrig tilldela något värdet undefined. De leder ofta till samma resultat, men har olika semantik och är inte helt identiska i effekt (tänk `if (member === null)`)

Reply:
Då håller jag med att null möjligen är rätt fel-värde. Personligen var min mentala model att user är definerat, mao. någon är inloggad, user "finns"==defined, eller undefined: någon är utloggad, user "finns inte"==undefined

Reply:
Hittade även detta, https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined,

enligt TypeScripts repo ska man inte använda sig av null alls i TS, men vi behöver ju inte följa det

Reply:
Denna var också rätt intressant https://basarat.gitbook.io/typescript/recap/null-undefined

Reply:
Intressant, det verkar som att TS har gått en annan väg en JS. Dock säger inte TS att man inte ska använda null alls i TS, utan att man inte ska använda det i kodbasen för tsc. Hittar inte jättemycket motiveringar för varför man borde använda undefined, utöver att Microsoft sagt det, men det verkar som att det kan lira bättre med nullable-typer i TS. Kör på med undefined isåfall. Det enda att hålla utkik för är om det kan clasha med graphql som kan ha null-värden.